ideascube/serveradmin/catalog.py

```diff
@@ -584,37 +584,55 @@ def _progress(*args):
             self._progress(' {}'.format(package.id), *args)
 
         filename = '{0.id}-{0.version}'.format(package)
+        urlparsed = urlparse(package.url)
 
         for cache in self._package_caches:
             path = os.path.join(cache, filename)
 
             if os.path.isfile(path):
                 if self._verify_sha256(path, package.sha256sum):
                     return path
+                if urlparsed.scheme in ['file', '']:
+                    try:
+                        shutil.copyfile(urlparsed.path, path)
+                    except Exception as error:
+                        print("Warning: Impossible to copy the package file"
+                              " {package.title}({package.url}).\n{error}\n"
+                              "Continue anyway without this remote."
+                              .format(package=package, error=error))
```

Comment:
You're copying a package, not a remote, so the message doesn't seem right.

Maybe something like `'Error copying the file: {}'.format(error)` to stay consistent with the download one?